src/config/ListenerConfig.ts

```diff
@@ -14,17 +14,28 @@
  * limitations under the License.
  */
 
+import {MembershipListener} from '../core/MembershipListener';
+
 /**
  * Configurations for LifecycleListeners. These are registered as soon as client started.
  */
 export class ListenerConfig {
-    lifecycle: Function[] = [];
+    lifecycleListeners: Function[] = [];
```

Comment:
nit: we should specify signature of these functions (I know that this code was present before this PR, but that should be a quick change). So, `Function[]` becomes something like `Array<(state: LifecycleState) => void>` (types in methods here should be updated as well).